File: front-end/src/app/modelo/permisos.ts
import { MotivoPermiso } from './MotivoPermiso';
import { Regimen } from './regimen';
import { TipoFormulario } from './tipoformulario';
import { TipoPermiso } from './tipopermiso';
import { Usuario } from './usuario';

export class Permisos {
  permId: number;
  permFechaEmision: Date;
  permObservacion: string;
  permFechaInicio: string;
  permFechaFin: string;
  permEstado: number;
  permHorasInicio: string;
  permHorasFin: string;
  permDocumento: string
  tiPeId: TipoPermiso;
  usuId: Usuario;
  tiFoId: TipoFormulario;
  motId: MotivoPermiso;

  constructor(
    permId?: number,
    permObservacion?: string,
    permFechaEmision?: Date,
    permFechaInicio?: string,
    permFechaFin?: string,
    permHorasInicio?: string,
    permHorasFin?: string,
    permEstado?: number,
    permDocumento?: string,
    tiPeId?: TipoPermiso,
    usuId?: Usuario,
    tiFoId?: TipoFormulario,
    motId?: MotivoPermiso,
  ) {
    this.permFechaEmision = permFechaEmision || new Date();
    this.permId = permId || 0;
    this.permObservacion = permObservacion || '';
    this.permFechaInicio = permFechaInicio || '';
    this.permFechaFin = permFechaFin || '';
    this.permHorasInicio = permHorasInicio || '';
    this.permHorasFin = permHorasFin || '';
    this.permEstado = permEstado || 0;
    this.permDocumento = permDocumento || '';
    this.tiPeId = tiPeId || new TipoPermiso();
    this.usuId = usuId || new Usuario();
    this.tiFoId = tiFoId || new TipoFormulario();
    this.motId = motId || new MotivoPermiso();
  }
}

File: back-end/app/src/main/java/com/senescyt/app/model/Permisos.java
package com.senescyt.app.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.io.Serial;
import java.io.Serializable;
import java.sql.Timestamp;
import java.time.LocalTime;
import java.util.Date;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "Permisos")
public class Permisos implements Serializable {

    @Serial
    private static final long serialVersionUID = 1L;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "permId")
    private Long permId;

    @Column(name = "permObservacion", length = 1000)
    private String permObservacion;
    @Column(name = "permFechaEmision")
    private Date permFechaEmision;
    @Column(name = "permFechaInicio")
    private String permFechaInicio;
    @Column(name = "permFechaFin")
    private String permFechaFin;

    /*estados del permiso
     * 1: Aprobado por Jefe General
     * 2: Aprobado por Jefe De Unidad
     * 3: En Espera
     * 4: Rechazado
     * */
    @Column(name = "permEstado")
    private int permEstado;
    @Column(name = "permHorasInicio")
    private String permHorasInicio;
    @Column(name = "permHorasFin")
    private String permHorasFin;
    @Column(name = "permDocumento", columnDefinition = "LONGTEXT")
    private String permDocumento;


    @ManyToOne
    @JoinColumn(name = "usuId", referencedColumnName = "usuId")
    private Usuario usuId;

    @ManyToOne
    @JoinColumn(name = "tiFoId", referencedColumnName = "tiFoId")
    private TipoFormulario tiFoId;

    @ManyToOne
    @JoinColumn(name = "motId", referencedColumnName = "motId")
    private MotivoPermiso motId;

    @ManyToOne
    @JoinColumn(name = "tiPeId", referencedColumnName = "tiPeId")
    private TipoPermiso tiPeId;

}
